application/models/user_access_token.go

```diff
@@ -102,3 +106,39 @@ func (u *UserAccessToken) GetOrganization() (Organization, error) {
 
 	return uOrg.Organization, nil
 }
+
+func createAccessTokenExpiry() time.Time {
+	lifetime := envy.Get("ACCESS_TOKEN_LIFETIME", "3600")
+
+	lifetimeSeconds, err := strconv.Atoi(lifetime)
+	if err != nil {
+		lifetimeSeconds = 28800
+	}
+
+	dtNow := time.Now()
+	futureTime := dtNow.Add(time.Second * time.Duration(lifetimeSeconds))
+
+	return futureTime
+}
+
+func createAccessTokenPart() string {
+	var alphanumerics = []rune("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+	tokenLength := 32
+	b := make([]rune, tokenLength)
+	for i := range b {
+		b[i] = alphanumerics[rand.Intn(len(alphanumerics))]
+	}
+
+	accessToken := string(b)
+
+	return accessToken
+}
+
+func (uat *UserAccessToken) Renew() error {
+	uat.ExpiresAt = createAccessTokenExpiry()
+	if err := DB.Update(uat); err != nil {
+		return fmt.Errorf("error renewing access token, %s", err)
```

Comment:
Somewhere in the chain of calls, this error should get logged.

Reply:
Done.